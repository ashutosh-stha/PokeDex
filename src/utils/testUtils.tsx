import React from 'react';
import {Provider} from 'react-redux';
import {Store} from 'redux';
import {init} from '@rematch/core';
import {render} from '@testing-library/react-native';
import {NavigationContainer} from '@react-navigation/native';

type WrapWithStore = {
  children: any;
};
type InitPropsType = {
  wrapper?: any;
};
const wrapWithStore = (store: Store) => {
  return ({children}: WrapWithStore) => (
    <Provider store={store}>
      <NavigationContainer>{children}</NavigationContainer>
    </Provider>
  );
};
class ComponentRenderer {
  private component;
  private models: any;
  private store: any;
  constructor(component: any) {
    this.component = component;
  }

  withModels(models: any) {
    this.models = models;
    return this;
  }
  render() {
    const initProps: InitPropsType = {};
    if (this.models) {
      initProps.wrapper = wrapWithStore(init({models: this.models}));
    }
    return render(this.component, initProps);
  }
}
export function componentRenderer(component: any) {
  return new ComponentRenderer(component);
}
