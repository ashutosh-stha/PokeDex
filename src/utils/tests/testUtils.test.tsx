import React from 'react';
import {Text} from 'react-native';
import {componentRenderer} from '../testUtils';

function MyComponent() {
  return <Text>Test Component</Text>;
}

describe('Component Renderer', () => {
  it('should render the component', () => {
    const renderResult = componentRenderer(<MyComponent />).render();
    expect(renderResult.getByText('Test Component')).toBeTruthy();
  });
});
