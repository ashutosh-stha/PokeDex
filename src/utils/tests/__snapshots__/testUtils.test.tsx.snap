// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Renderer should render the component 1`] = `
<Text>
  Test Component
</Text>
`;
